locustempus/main/views.py

```diff
@@ -71,3 +74,22 @@ def get_success_url(self) -> str:
 class CourseRosterView(LoggedInFacultyMixin, DetailView):
     model = Course
     template_name = 'main/course_roster.html'
+
+    def post(self, request, *args, **kwargs):
+        """Toggles the role of course users between faculty/student"""
+        user = get_object_or_404(
+            User, id=request.POST.get('user_id', None))
+        course = get_object_or_404(Course, id=kwargs.get('pk', None))
+
+        if course.is_true_faculty(user):
+            course.faculty_group.user_set.remove(user)
+            msg = u'{} is now a student'.format(user_display_name(user))
+        elif course.is_member(user):
+            course.faculty_group.user_set.add(user)
+            msg = u'{} is now faculty'.format(user_display_name(user))
+        else:
+            msg = u'{} is not a user in this course'.format(
+                user_display_name(user))
+
+        messages.add_message(request, messages.INFO, msg)
+        return HttpResponseRedirect('')
```

Comment:
I think this needs to redirect back to `course-roster-view`.

Reply:
Yeah, I was being lazy. The docs say that you can use a relative URL, so an empty string brings you back to the same page.
It's better to make it more specific though. I think it's made the tests more clear by specifying the expected path.

Reply:
Note that in testing in my local dev env, this redirect did not work.